src/common/helpers/email/email.test.js

```diff
@@ -0,0 +1,206 @@
+import {
+  sendEmailToApplicant,
+  sendEmailToCaseWorker
+} from '../../../common/connectors/notify/notify.js'
+import { statusCodes } from '../../../common/constants/status-codes.js'
+import {
+  compressFile,
+  fetchFile
+} from '../../../common/helpers/file/file-utils.js'
+import { emailApplicationHandler } from './email.js'
+
+const mockedFileProps = {
+  filename: 'file-name.pdf'
+}
+
+jest.mock('../../../common/connectors/notify/notify.js')
+jest.mock('../../../common/helpers/file/file-utils.js')
+jest.mock('../../../common/helpers/email-content/email-content.js', () => ({
+  generateEmailContent: jest.fn().mockReturnValue('Case worker email content'),
+  getFileProps: jest.fn().mockReturnValue({
+    filename: 'file-name.pdf'
+  })
+}))
+jest.mock('../../../common/connectors/sharepoint/sharepoint.js', () => ({
+  uploadFile: jest.fn()
+}))
+
+const mockFetchFile = /** @type {jest.Mock} */ (fetchFile)
+const mockCompressFile = /** @type {jest.Mock} */ (compressFile)
+
+const testEmail = 'test@example.com'
+const testFullName = 'Name Surname'
+const testReferenceNumber = 'TB-1234-5678'
+
+const emailQuestion = {
+  question: 'emailAddress',
+  questionKey: 'emailAddress',
+  answer: {
+    type: 'email',
+    value: 'test@example.com',
+    displayText: 'test@example.com'
+  }
+}
+const fullNameQuestion = {
+  question: 'fullName',
+  questionKey: 'fullName',
+  answer: {
+    type: 'name',
+    value: { firstName: 'Name', lastName: 'Surname' },
+    displayText: testFullName
+  }
+}
+const fileQuestion = {
+  question: 'upload-plan',
+  questionKey: 'upload-plan',
+  answer: {
+    type: 'file',
+    value: {
+      skipped: false,
+      path: 'file.pdf'
+    },
+    displayText: 'file.pdf'
+  }
+}
+
+const mockLogger = { info: jest.fn(), warn: jest.fn(), error: jest.fn() }
+
+const mockRequest = {
+  logger: mockLogger,
+  payload: {
+    sections: [
+      {
+        section: 'licence',
+        sectionKey: 'licence',
+        questionAnswers: [emailQuestion, fullNameQuestion]
+      }
+    ]
+  }
+}
+
+const mockRequestWithFile = {
+  logger: mockLogger,
+  payload: {
+    sections: [
+      {
+        section: 'licence',
+        sectionKey: 'licence',
+        questionAnswers: [emailQuestion, fullNameQuestion]
+      },
+      {
+        section: 'biosecurity-map',
+        sectionKey: 'biosecurity-map',
+        questionAnswers: [fileQuestion]
+      }
+    ]
+  }
+}
+
+const mockFile = {
+  file: 'mock-file',
+  contentType: 'application/pdf'
+}
+
+describe('emailApplicationHandler', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
```

Comment:
I believe you need an `afterAll(jest.restoreAllMocks)` here as well, because otherwise those modules will be remain mocked for other test runs.